test/functional_execution_context/tests/execution_context.ts

```diff
@@ -0,0 +1,372 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import type { Ecs, KibanaExecutionContext } from 'kibana/server';
+
+import Fs from 'fs/promises';
+import Path from 'path';
+import { isEqual } from 'lodash';
+import type { FtrProviderContext } from '../ftr_provider_context';
+
+const logFilePath = Path.resolve(__dirname, '../kibana.log');
+
+export default function ({ getService, getPageObjects }: FtrProviderContext) {
+  const PageObjects = getPageObjects(['common', 'dashboard', 'header', 'home']);
+  const retry = getService('retry');
+
+  async function assertLogContains(
+    description: string,
+    predicate: (record: Ecs) => boolean
+  ): Promise<void> {
+    // logs are written to disk asynchronously. I sacrificed performance to reduce flakiness.
+    await retry.waitFor(description, async () => {
+      const logsStr = await Fs.readFile(logFilePath, 'utf-8');
+      const normalizedRecords = logsStr
+        .split('\n')
+        .filter(Boolean)
+        .map((s) => JSON.parse(s));
+
+      return normalizedRecords.findIndex(predicate) !== -1;
+    });
+  }
+
+  function isExecutionContextLog(
+    record: string | undefined,
+    executionContext: KibanaExecutionContext
+  ) {
+    if (!record) return false;
+    try {
+      const object = JSON.parse(record);
```

Comment:
I parse logs to simplify reading and avoid flakiness if someone changes the order of properties in the future.